File: readability/readability.py
from cs50 import get_string


def main():
    # I ask the user some string input
    Phrase = get_string("Type something here: ")
    
    # Call some procedure that  gives the grade
    Grade = Quantity(Phrase)

    PrintGradeClasification(round(Grade))
    
    
def Quantity(Phrase):
    lengt = len(Phrase)
    # I need the exact words
    Words = QuantityWords(Phrase, lengt)
    # I need the exact Letters
    Letters = QuantityLetters(Phrase, lengt)
    # I need the exact sentences
    Sentences = QuantitySentences(Phrase, lengt)
    # I return te exact Grade of the Paragraph
    G = (0.0588 * (100 * (Letters / Words)) - 0.296 * (100 * (Sentences / Words)) - 15.8)
    return G
    
    
def QuantityLetters(Phrase, lengt):
    # I delcare the local variables
    i = 0
    L = 0
    # I need iterate for every alphabetic char in the paragraph
    while i != lengt:
        # If char = alphabetic, then add 1
        if (Phrase[i].isalpha()):
            L += 1
        # Add to mover foward the string
        i += 1
    return L
            
            
def QuantityWords(Phrase, lengt):
    # I delcare the local variables
    i = 0
    W = 1
    # I need iterate for every spaces in the paragraph
    while i != lengt:
        # If there is a spaces that meas thats a new word
        if (Phrase[i] == ' '):
            W += 1
        # Add to move foward the string
        i += 1
    return W


def QuantitySentences(Phrase, lengt):
    # I declare the local variables
    i = 0
    S = 0
    while i != lengt:
        # If there is a: . ! or ? == end of the sentences
        if ((Phrase[i] == '.') or (Phrase[i] == '!') or (Phrase[i] == '?')):
            S += 1
        # Add to move foward the string
        i += 1
    return S
    
    
def PrintGradeClasification(Grade):
    
    if (Grade >= 16):
        print("Grade 16+")
    elif (Grade <= 1):
        print("Before Grade 1")
    else:
        print(f"Grade {Grade}")
    
    
if __name__ == "__main__":
    main()

File: pset2/problems2/Readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>
#include <stdlib.h>

float Quantity(string Paragraph);
int QuantityLetters(string Paragraph);
int QuantityWords(string Paragraph);
int QuantitySentences(string Paragraph); 

int main(void)
{
    string Paragraph = get_string("text: ");
    float Grade = Quantity(Paragraph);
    if ((round(Grade) > 1) && (round(Grade) < 16))
    {
        printf("Grade %0.f\n", round(Grade));
    }
    else if (round(Grade) < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (round(Grade) > 16)
    {
        printf("Grade 16+\n");
    }
}


float Quantity(string Paragraph)
{
    float W = QuantityWords(Paragraph);
    float L = QuantityLetters(Paragraph);
    float S = QuantitySentences(Paragraph);
    double index = 0.0588 * (100 * (L / W)) - 0.296 * (100 * (S / W)) - 15.8; 
    return index;
}

int QuantitySentences(string Paragraph)
{
    int Sentences = 0;
    for (int b = 0, m = strlen(Paragraph) ; b < m; b++)
    {
        if ((Paragraph[b] == '.') || (Paragraph[b] == '!') || (Paragraph[b] == '?'))
        {
            Sentences = Sentences + 1.0;
        }
    }
    return Sentences;
}


int QuantityWords(string Paragraph)
{
    int Words = 1;
    for (int a = 0, n = strlen(Paragraph); a < n; a++)
    {
        if (Paragraph[a] == ' ')
        {
            Words = Words + 1.0;
        }
    }
    return Words;
}

int QuantityLetters(string Paragraph)
{
    int Letters = 0;
    for (int i = 0, N = strlen(Paragraph); i < N ; i++)
    {
        if (isalpha(Paragraph[i]))
        {
            Letters = Letters + 1.0;
        }
    }
    return Letters;
}